### PR TITLE
link_overwrite: allow to overwrite files from alias

### PR DESCRIPTION
**Pull request by iMichka**
See https://github.com/Homebrew/homebrew-core/pull/54912#issuecomment-653893645
Formulary.factory("python") points to python@3.8, which breaks link_overwrite
for that case.

This changes checks if the formula is an alias, so that we can still override
the files during installation.

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: Library/Homebrew/formula.rb [hunk 1]
```diff
@@ -1112,13 +1112,15 @@ def link_overwrite?(path)
...
+        f = Formulary.factory(keg.name)
       rescue FormulaUnavailableError
         # formula for this keg is deleted, so defer to allowlist
       rescue TapFormulaAmbiguityError, TapFormulaWithOldnameAmbiguityError
         return false # this keg belongs to another formula
       else
-        return false # this keg belongs to another formula
+        # this keg belongs to another formula
+        # but it is not an alias
+        return false unless f.aliases.include? keg.name
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula.rb:1123]:
```suggestion
        # this keg belongs to another unrelated formula
        return false unless (f.aliases + f.oldname).include?(keg.name)
```

**Comment by MikeMcQuaid** [comment]
Makes sense, let's give it a go!

**Comment by iMichka** [comment]
Cool. I'll rebase my python 3.8 PR today and will make a final test. We will need to cut a new release too before shipping it.

**Comment by MikeMcQuaid** [comment]
@iMichka Sounds good. Happy to cut a release pretty much as soon as you're ready.

**Pull request merged**